### PR TITLE
Add Definitions

### PR DESCRIPTION
**Pull request by ianswett**
Add a separate definitions section, similar to the transport doc.

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+
+Sender:
+
+: The endpoint sending QUIC packets.
+
+Receiver:
+
+: The endpoint receiving QUIC packets and sending acknowledgements in
+  response.
+
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: I don't think we need the above ones. In fact they may be more confusing than useful (sender of acks, for instance)

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+ACK-only:
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: I think this includes PADDING. Also, you'll want to fix all uses in the doc (ack-only, ack only)

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+
+: Any packet containing only an ACK or ACK_ECN frame. The rest of this
+  document uses "ACK frames" to refer to both ACK and ACK_ECN frames.
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: This second sentence doesn't seem to fit in this definition. I would separate it out into a new definition for now.

File: draft-ietf-quic-recovery.md [hunk 4]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+
+In-flight:
+
+: Packets are considered in-flight because they have been sent
+  and neither acknowledged or considered lost, and they are not
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: s/or/nor/

File: draft-ietf-quic-recovery.md [hunk 5]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+  response.
+
+ACK-only:
+
+: Any packet containing only an ACK or ACK_ECN frame. The rest of this
+  document uses "ACK frames" to refer to both ACK and ACK_ECN frames.
+
+In-flight:
+
+: Packets are considered in-flight because they have been sent
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: s/because/when/

File: draft-ietf-quic-recovery.md [hunk 6]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+  and neither acknowledged or considered lost, and they are not
+  ACK-only.
+
+Retransmittable:
+
+: Packets that contain frames besides ACK, ACK_ECN, or PADDING elicit
+  an ACK from the receiver and are called retransmittable.
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: I think you should define retransmittable frames, and packets containing a retransmittable frame as a retransmittable packet.

File: draft-ietf-quic-recovery.md (hunk 4 above)
Comment (ianswett) [reply draft-ietf-quic-recovery.md]: Hmm, I think it's or.  But I changed considered to declared.

File: draft-ietf-quic-recovery.md [hunk 7]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+
+ACK frames:
+
+: ACK frames refer to both ACK and ACK_ECN frames in this
+  document.
+
+ACK-only:
+
+: Any packet containing only an ACK frame.
+
```
Comment (mikkelfj) [inline draft-ietf-quic-recovery.md]: Suggest being specific about ACK and ACK_ECN here

File: draft-ietf-quic-recovery.md [hunk 8]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+Retransmittable Frames:
+
+: All frames besides ACK, ACK_ECN, or PADDING are considered
+  retransmittable.
+
+Retransmittable Packets:
+
+: Packets that contain retransmittable frames elicit an ACK from
+  the receiver and are called retransmittable packets.
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md]: Both packets and frames aren't classified as retransmittable or not.  It would be better if this used a different label.

Comment (janaiyengar) [reply draft-ietf-quic-recovery.md]: @martinthomson : I don't understand your comment. Can you rephrase?

Comment (martinthomson) [reply draft-ietf-quic-recovery.md]: The notion that a packet is retransmittable is a misnomer because it creates the impression that you can retransmit the packet.  Similarly, we don't retransmit frames either.  The distinction that is relevant to your uses of these terms is not whether the packet is able to be retransmitted, but whether the packet contains frames other than ACK/ACK_ECN/PADDING.

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: I opened issue #1595 for splitting up the terms, but I'd prefer do that separately.

File: draft-ietf-quic-recovery.md [hunk 9]
```diff
@@ -79,13 +79,40 @@ recovery, and where applicable, attributes the TCP equivalent in RFCs,
...
+  document.
+
+ACK-only:
+
+: Any packet containing only an ACK or ACK_ECN frame.
+
+In-flight:
+
+: Packets are considered in-flight when they have been sent
+  and neither acknowledged or declared lost, and they are not
```
Comment (MikeBishop) [inline draft-ietf-quic-recovery.md]: Late to the party, but:

Either/or, but Neither/nor

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: Thanks Mike, I fixed it in master
